### DISCUSSION TITLE
Getting 403 with custom API authentication even after the successful authentication

### DISCUSSION THREAD

**Discussion by pragyak412** [apache/airflow#20676] [Q&A]
Hello Team,
I wanted to use custom API authentication with my airflow API hence I followed this link https://airflow.apache.org/docs/apache-airflow/2.2.0/security/api.html#roll-your-own-api-authentication, and wrote a custom library. But even after successfully getting authenticated, I am getting a 403 error.
Let me know what all parameters need to be changed to make sure custom authentication works with the API.
```
[api]
auth_backend =<nameOfCustomAuth>
```

**Comment by brandonmino** [comment]
@pragyak412 did you resolve this? I am still having this issue myself on Airflow version 2.4.1.

**Comment by potiuk** [comment]
It's impossible to help without any logs or details I am afraid. You might also simply try to debug what happens using debugger. Writing custom authentication usually means that you need to be ready to run and debug your code.

**Comment by eduardchai** [comment]
I encountered the same issue.

@potiuk I created a custom module for testing that exactly the same as https://github.com/apache/airflow/blob/main/airflow/api/auth/backend/default.py. And add the configuration via env var:

```
"AIRFLOW__API__AUTH_BACKENDS"                            = "custom.airflow.api.auth.backend.default"
```

Any recommendations on how to debug this issue? Thank you

**Reply by potiuk** [reply]
Enable debug logging, see if you can see some strange things. the usual drill. Possibly look at the FAB documentation (FAB is providing the authentication for us). Webserver gnerally prints the reasons for returning 403 errors and if you increase logging to debug (see our docs how to do it) - you will likely see some suspicious messages. 

You can also - as usual for Python apps - step-debug with debugger - maybe it will give more insight/information?
